packages/@react-spectrum/s2/src/TagGroup.tsx

```diff
@@ -83,42 +95,179 @@ const helpTextStyles = style({
   cursor: 'text'
 });
 
-function TagGroup<T extends object>(
-  {
+const InternalTagGroupContext = createContext<TagGroupProps<any>>({});
+
+function TagGroup<T extends object>(props: TagGroupProps<T>, ref: DOMRef<HTMLDivElement>) {
+  props = useFormProps(props);
+  let {onRemove} = props;
+  return (
+    <InternalTagGroupContext.Provider value={{onRemove}}>
+      <CustomTagGroup {...props} forwardedRef={ref}>
+        <TagList style={{display: 'flex', gap: 4}}>
+          {props.children}
+        </TagList>
+      </CustomTagGroup>
+    </InternalTagGroupContext.Provider>
+  );
+}
+
+/** Tags allow users to categorize content. They can represent keywords or people, and are grouped to describe an item or a search request. */
+let _TagGroup = /*#__PURE__*/ (forwardRef as forwardRefType)(TagGroup);
+export {_TagGroup as TagGroup};
+
+interface CustomTagGroupProps<T> extends TagGroupProps<T> {
+  forwardedRef: DOMRef<HTMLDivElement>
+}
+
+function CustomTagGroup<T>(props: CustomTagGroupProps<T>) {
+  return (
+    // @ts-ignore how do i fix this one?
```

Comment:
That's because `children` can also be a function for dynamic collections. So you need `content={<Collection {...props} />}`